app/controllers/concerns/foreman_openscap/hosts_controller_extensions.rb

```diff
@@ -10,5 +10,38 @@ def process_hostgroup
       @openscap_proxy = @hostgroup.openscap_proxy
       super
     end
+
+    def select_multiple_openscap_proxy
+      find_multiple
+    end
+
+    def update_multiple_openscap_proxy
+      if (id = params['smart_proxy']['id'])
+        find_multiple
+        smart_proxy = ::SmartProxy.find(id)
+        @hosts.each do |host|
+          host.openscap_proxy = smart_proxy
+          host.save!
+        end
+        notice _("Updated hosts: Assigned with OpenSCAP Proxy: %s") % smart_proxy.name
+        redirect_to hosts_path
+      else
+        error _('No OpenSCAP Proxy selected.')
+        redirect_to(select_multiple_openscap_proxy_hosts_path)
+      end
+    end
+
+    private
+
+    def action_permission
+      case params[:action]
+      when 'select_multiple_openscap_proxy'
```

Comment:
in `engine.rb`, this one is under `:edit_hosts`, so maybe we should keep it consistent?

Reply:
Switched to using `:edit` for consistency.

Reply:
Thanks, you can actually branch on multiple conditions, could you get rid of one branch by changing it to the following please? My last comment, cross my heart :)

```ruby
def action_permission
    case params[:action]
    when 'select_multiple_openscap_proxy', 'update_multiple_openscap_proxy'
      :edit
    else
      super
    end
  end
```

Reply:
Change made